### PR TITLE
Ecom UI

### PR DESCRIPTION
**Pull request by vrx29**
Live Preview : https://gimesto-ecom.netlify.app/

This PR contains UI for Ecommerce App this doesn't have functionality.

--|--**components**
--|--|--Dropdown
--|--|--Footer
--|--|--Navbar
--|--|--ProductCard
--|--**pages**
--|--|--404
--|--|--Cart
--|--|--Home
--|--|--Login
--|--|--Products
--|--|--SignUp
--|--|--WishList

### CODE REVIEW

File: src/pages/Cart/components/CartCard/CartCard.jsx [hunk 1]
```diff
@@ -0,0 +1,32 @@
+import React from "react";
+import { HeartIcon } from "../../../../assets/icons";
+
+export function CartCard({ product }) {
+  return (
+    <div className="cart-item">
+      <div className="cart-item-img">
+        <img src={product.img} alt={product.name} loading="lazy" />
```
Comment (BalajiHarinath) [inline src/pages/Cart/components/CartCard/CartCard.jsx]: You can de-structure the product details in the function parameter itself and use them.

File: src/pages/Home/Home.jsx [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+
+export function Home() {
+  const [products, setProducts] = useState([]);
+
+  useEffect(() => {
+    (async function () {
+      const res = await axios.get("/api/products");
+      const data = await res.data;
+      setProducts(data.products);
+      console.log(products);
```
Comment (BalajiHarinath) [inline src/pages/Home/Home.jsx]: can implement these in try catch block to handle the errors.

**Review by BalajiHarinath — APPROVED** [review]
Appreciation : Your folder structure is good and your UI is awesome.

File: src/pages/Cart/cart.css [hunk 3]
```diff
@@ -0,0 +1,129 @@
...
+  padding: 2rem 1rem;
+  justify-content: center;
+}
+.cart-left-cont {
+  display: flex;
+  flex-direction: column;
+  flex-grow: 1;
+}
+.cart-right-cont {
+  width: 340px;
```
Comment (abhisheksahani-nova) [inline src/pages/Cart/cart.css:13]: Instead of px using rem  will help in mobile responsive too

File: src/pages/Home/Home.jsx (hunk 2 above)
Comment (abhisheksahani-nova) [inline src/pages/Home/Home.jsx]: Remove console.log once used and tested the code

File: src/pages/Products/Products.jsx [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+import { ProductCard } from "../../components";
+import { Filters } from "./components";
+import axios from "axios";
+
+export function Products() {
+  const [products, setProducts] = useState([]);
+
+  useEffect(() => {
+    (async function () {
+      const res = await axios.get("/api/products");
```
Comment (abhisheksahani-nova) [inline src/pages/Products/Products.jsx]: Suggestion : -
 
- You have done the same api call in the Home.jsx to get products data , doing the same api call twice is not a good practice and also you are creating products , setProducts state twice  , it affects performance , instead yu can store the products into the context , you can discuss this with TA and MA as well .

**Review by abhisheksahani-nova — CHANGES_REQUESTED** [review]
Look in the changes mention, else everything looks subtle , good work

File: src/pages/Home/Home.jsx (hunk 2 above)
Comment (vrx29) [reply src/pages/Home/Home.jsx]: will fix this

File: src/pages/Cart/components/CartCard/CartCard.jsx (hunk 1 above)
Comment (vrx29) [reply src/pages/Cart/components/CartCard/CartCard.jsx]: okay will try this

File: src/pages/Cart/cart.css (hunk 3 above)
Comment (vrx29) [reply src/pages/Cart/cart.css:13]: I need it to be fixed.

File: src/pages/Products/Products.jsx (hunk 4 above)
Comment (vrx29) [reply src/pages/Products/Products.jsx]: This is just the UI component I haven't created context yet. Will move it there once I create the context

Comment (abhisheksahani-nova) [reply src/pages/Products/Products.jsx]: Great

**Review by abhisheksahani-nova — APPROVED** [review]

**Pull request merged**